### PR TITLE
Use latest Nix version in the setup script

### LINKED ISSUE #4035: Update Nix version installed by the setup script
People are starting to have trouble with this -> https://github.com/holochain/holochain/blob/holochain.github.io/setup.sh#L7

We should move everybody up to the latest version of Nix. I can confirm that it works fine with the scaffolding and my configuration but we might need to tweak how we configure Nix on people's systems.

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Closes #4035

Script uses `curl` to fetch Nix. Not available by default on Ubuntu! Use `wget` instead which is pretty much universally available?

### TODO:
- [ ] CHANGELOGs updated with appropriate info
- [ ] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 set -e
 
-# bump this to the most recent version when the following bugs have been resolved
-# - https://github.com/NixOS/nix/issues/7984
```
Comment (ThetaSinner) [inline setup.sh:6]: Bug was fixed in March 2023...

**Comment by ThetaSinner** [comment]
I ran this through on a Ubuntu VM. Certainly works to get started. I've been running v2.21 for a while and upgraded to v2.23 a couple of days ago. No issues noticed.

File: .github/workflows/test-holonix-setup.yml [hunk 2]
```diff
@@ -69,7 +69,7 @@ jobs:
           # TODO replace with nix-cache-check
           function do_nix_build {
             nix build -L --refresh -j0 \
-              --override-input versions "github:holochain/holochain?dir=versions/0_1" \
+              --override-input versions "github:holochain/holochain?dir=versions/0_3" \
```
Comment (ThetaSinner) [inline .github/workflows/test-holonix-setup.yml:72]: `0_1` is no longer cached, not such a good test any more :)

**Comment by jost-s** [comment]
> Script uses curl to fetch Nix. Not available by default on Ubuntu! Use wget instead which is pretty much universally available?

`wget` is not available on macos by default =(

**Review by jost-s — APPROVED** [review]

**Comment by ThetaSinner** [comment]
> wget is not available on macos by default =(

Haha, of course it isn't. There really needs to be one CLI tool for doing HTTP requests that vendors agree to package. It would help a great deal! :)

**Comment by jost-s** [comment]
Yes, it should be part of POSIX

**Pull request merged**